### ISSUE TITLE
Map on Lst<T> performs in a different order

### ISSUE THREAD

**Issue by StanJav** [louthy/language-ext#373]
Using `Map` with `Lst<T>` goes through the list in a different order. 

Is it a bug or just an unexpected behaviour? 

Example:
```
var myList = List(1, 2, 3, 4, 5).Map(i =>
{
    Console.WriteLine(i);
    return i;
});
```

Output:
```
3
2
1
5
4
```

Note that the result `myList` is in a correct order (1,2,3,4,5).

**Comment by bender2k14** [comment]
Not sure if it is a bug, but that behavior is because of [this line](https://github.com/louthy/language-ext/blob/master/LanguageExt.Core/DataTypes/List/Lst.Internal.cs#L531).

The example code uses an internal data structure that is a tree.  The current behavior (that follows from the order of the arguments in the method call on the linked line) is a [pre-order traversal](https://en.wikipedia.org/wiki/Tree_traversal#Pre-order) of this tree data structure.  @StanJav was expecting an [in-order traversal](https://en.wikipedia.org/wiki/Tree_traversal#In-order), which can be achieved by swapping the third and fourth arguments.

I can create a PR to make this change.

**Comment by louthy** [comment]
@StanJav This is fixed in `v2.1.11`

I guess if you're writing pure code it shouldn't really matter, but the 'real world' isn't always as easy, so I agree it needed fixing.
